helix-term/src/ui/editor.rs

```diff
@@ -618,12 +650,29 @@ impl EditorView {
                     (true, false) => gutter_selected_style_virtual,
                 };
 
+                let mut doc_line = Some(pos.doc_line);
+
+                if let Some(current_context) = context
+                    .as_ref()
+                    .and_then(|c| c.iter().find(|n| n.visual_line == pos.visual_line))
+                {
+                    doc_line = if current_context.indicator.is_some() {
+                        None
+                    } else {
+                        Some(text.byte_to_line(current_context.byte_range.start))
+                    };
+                }
+
                 if let Some(style) =
-                    gutter(pos.doc_line, selected, pos.first_visual_line, &mut text)
+                    gutter(doc_line, selected, pos.first_visual_line, &mut text_to_draw)
                 {
```

Comment:
Why not drop the `doc_line = Some(pos.doc_line)` but instead only call `gutter` when it is some, so we don't have to pass `Some(pos.doc_line)` but we can just pass `pos.doc_line` like we used to do?

Reply:
I am not sure if i understand you correctly, but here is the reason why i pass an `Option`:
For the indicator, since the indicator has no line-number, i don't want to draw anything on the gutter for it.

Reply:
Ah, I didn't realize it is for the gutter ruler, didn't look closely and assume all of them have a line number.